### PR TITLE
Async grpc server gsv1 support

### LINKED ISSUE #259: [Enhancement] Add support for GoalState V1 in the gRPC server.
After implementing the async server in the ACA, it has been unable to receive GoalState V1 from the DPM. This is because the current code only supports GoalState V2. 

As GoalState V1 will still be relevant in the forseeable future, we need to support it too.

### PR DESCRIPTION
**Pull request by zzxgzgz**
This PR does the following:

1. Added support for `PushGoalStatesStream` in the async gRPC server, which makes ACA able to receive GoalStateV1. Closes #259 
2. Refactored the code in the async gRPC server, so that it will be easier to add more rpc calls in the future.

### CODE REVIEW

File: include/aca_grpc.h [hunk 1]
```diff
@@ -21,55 +21,88 @@
...
 using grpc::Status;
 
 class GoalStateProvisionerAsyncServer {
   public:
+  /*
+    Base class that represents a gRPC call.
+    When you have a new kind of rpc, add the corresponding enum to CallType
+  */
+  struct AsyncGoalStateProvionerCallBase {
+    enum CallType { PUSH_NETWORK_RESOURCE_STATES, PUSH_GOAL_STATE_STREAM };
```
Comment (xieus) [inline include/aca_grpc.h:42]: Very good to have a small enum like this. What is the difference between the two? A short comment would help.

File: include/aca_grpc.h [hunk 2]
```diff
@@ -21,55 +21,88 @@
...
+    enum CallStatus { INIT, SENT };
+    CallStatus status_;
+    CallType type_;
+    grpc::ServerContext ctx_;
+  };
+
+  //  struct for PushNetworkResourceStates, which is a unary gRPC call
+  //  when adding a new unary rpc call, create a new struct just like PushNetworkResourceStatesAsyncCall
+  struct PushNetworkResourceStatesAsyncCall : public AsyncGoalStateProvionerCallBase {
```
Comment (xieus) [inline include/aca_grpc.h:51]: Why struct vs. class?

**Review by xieus — COMMENTED** [review]
@zzxgzgz a few minor comments.

File: src/comm/aca_grpc.cpp [hunk 3]
```diff
@@ -48,117 +47,278 @@ GoalStateProvisionerAsyncServer::ShutDownServer()
...
-          status_ = READY_TO_WRITE;
-          break;
-      }
-      case DONE:
-      {
-          ACA_LOG_DEBUG("Killing the stream (Async GRPC)\n");
-          delete this;
-          break;
-      }
+  while (true) {
```
[the remark this answers is not in the document]
Comment (zzxgzgz) [reply src/comm/aca_grpc.cpp]: Added a boolean variable, which will stop the server from stucking in this while loop, when the server destructor is called.

File: include/aca_grpc.h (hunk 1 above)
Comment (zzxgzgz) [reply include/aca_grpc.h:42]: Added

File: include/aca_grpc.h (hunk 2 above)
Comment (zzxgzgz) [reply include/aca_grpc.h:51]: I was following an online example, in which it uses struct. I guess we can use either one of them.

**Review by xieus — APPROVED** [review]
LGTM.

**Pull request merged**